csharp/src/Drivers/Apache/Spark/SparkDatabase.cs

```diff
@@ -30,9 +32,30 @@ public SparkDatabase(IReadOnlyDictionary<string, string> properties)
 
         public override AdbcConnection Connect(IReadOnlyDictionary<string, string>? properties)
         {
-            SparkConnection connection = new SparkConnection(this.properties);
+            return ((IMockingDatabase<TCLIService.IAsync>)this).Connect(properties, mock: default);
+        }
+
+        MockingConnection<TCLIService.IAsync> IMockingDatabase<TCLIService.IAsync>.Connect(IReadOnlyDictionary<string, string>? properties, MockDataSourceBase<TCLIService.IAsync>? mock)
+        {
+            IReadOnlyDictionary<string, string> combinedProperties = MergeDictionaries(this.properties, properties);
+            SparkConnection connection = new(combinedProperties, mock);
             connection.OpenAsync().Wait();
             return connection;
         }
+
+        private static IReadOnlyDictionary<TKey, TValue> MergeDictionaries<TKey, TValue>(params IReadOnlyDictionary<TKey, TValue>?[] dictionaries)
```

Comment:
Amazingly, this isn't part of .NET to merge two or more dictionaries in the case there are duplicates.